packages/models/src/actions/code.ts

```diff
@@ -14,61 +11,90 @@ export enum CodeActionType {
     UNGROUP = 'ungroup',
 }
 
-interface BaseCodeAction {
-    type: CodeActionType;
-    location: ActionElementLocation;
-    uuid: string;
-}
+const BaseCodeActionSchema = z.object({
+    type: z.nativeEnum(CodeActionType),
+    location: ActionElementLocationSchema,
+    uuid: z.string(),
+});
 
-export interface IndexMoveLocation extends ActionElementLocation {
-    position: InsertPos.INDEX;
-    targetSelector: string;
-    index: number;
-}
+export const IndexMoveLocationSchema = ActionElementLocationSchema.extend({
+    position: z.literal(InsertPos.INDEX),
+    targetSelector: z.string(),
+    index: z.number(),
+});
 
-export interface CodeMove extends BaseCodeAction {
-    selector: string;
-    type: CodeActionType.MOVE;
-    location: MoveActionLocation;
-    childTemplateNode: TemplateNode;
-}
+export const CodeMoveSchema = BaseCodeActionSchema.extend({
+    selector: z.string(),
+    type: z.literal(CodeActionType.MOVE),
+    location: ActionElementLocationSchema.extend({
```

Comment:
Is there no way to preserve MoveActionLocation? 
This is an important type and used often.

Reply:
Pushing a fix for this